src/plugin/luapi_application.h

```diff
@@ -475,6 +476,164 @@ static int applib_changeBackgroundPdfPageNr(lua_State* L) {
+ * {
+ *   "size" = string
+_*   "fill" = bool
+ *   "opacity" = integer
+ * }
+ *
+ * Example: local penInfo = app.getPenInfo()
+ */
+static int applib_getPenInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string size = toolSizeToString(toolHandler->getPenSize());
+    int fillOpacity = toolHandler->getPenFill();
+    bool fill = toolHandler->getPenFillEnabled();
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fill");
+    lua_pushboolean(L, fill);
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fillOpacity");
+    lua_pushinteger(L, fillOpacity);
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+/*
+ * Returns a table encoding all info on the highlighter in a Lua table of the shape
+ * {
+ *   "size" = string
+_*   "fill" = bool
+ *   "opacity" = integer
+ * }
+ *
+ * Example: local highlighterInfo = app.getHighlighterInfo()
+ */
+static int applib_getHighlighterInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string size = toolSizeToString(toolHandler->getHighlighterSize());
+    int fillOpacity = toolHandler->getHighlighterFill();
+    bool fill = toolHandler->getHighlighterFillEnabled();
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fill");
+    lua_pushboolean(L, fill);
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fillOpacity");
+    lua_pushinteger(L, fillOpacity);
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+/*
+ * Returns a table encoding all info on the eraser in a Lua table of the shape
+ * {
+ *   "type" = string
+ *   "size" = string
+ * }
+ *
+ * Example: local eraserInfo = app.getEraserInfo()
+ */
+static int applib_getEraserInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string type = eraserTypeToString(toolHandler->getEraserType());
+    std::string size = toolSizeToString(toolHandler->getEraserSize());
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "type");
+    lua_pushstring(L, type.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+
+/*
+ * Returns a table encoding all info on the active tool in a Lua table of the shape
+ * {
+ *   "tool" = string
+ *   "color" = integer
+ *   "fillOpacity" = integer,
+ *   "drawingType" = string
+ *   "lineStyle" = string
+ *   "thickness" = number
+ * }
+ *
+ * Example: local activeToolInfo = app.getActiveToolInfo()
```

Comment:
Although perhaps better suited for the website, documenting some of the possible values for 'tool', 'color', etc. could be helpful here.
```suggestion
 * Example 1: local activeToolInfo = app.getActiveToolInfo()
 * Example 2: app.getActiveToolInfo()["tool"]
 *  - Returns the name of the active tool. See /src/control/ToolEnums.cpp#toolTypeToString for possible values.
```